Powerup pickup

amtrang opened the pull request.

amtrang committed:
Equipable Entity Prototype 1

Player can check if a weapon or ability is close enough to be checked.

amtrang committed:
Equipable Entity: Weapons 1

Made it possible to grab a new weapon for the player when they are near it and press the Interact button.

amtrang committed:
Weapon Swap Issue

When the player tries to replace their current weapon with a new one when they can't carry too many weapons, the new weapon goes away from the weapon handler, and the old weapon does not go to the new weapon's position.

CristionD committed:
Mess with weapon animator (bug still present)

CristionD committed:
Remedy position bug when replacing current weapon

amtrang committed:
Equipable Entity Proto. Finished

Use raycast to collect weapons and abilities. Player can only carry a set number of weapons and abilities before dropping them.

CristionD reviewed (commented):
I request 2 changes:
- Do not cycle through all abilities when the Player takes a new ability.
- Ensure the Player does not take duplicate abilities.

amtrang committed:
Ability Handler Adjustments 1

Ability handler no longer loops through all of the player's abilities to enable them every time pick up a new one. Also, it checks if the player already has an ability when picking up a new one.

CristionD committed:
Replace `GameObject` references with `IWeapon` and `EquippableEntity` references.
